src/theory/arith/arith_rewriter.cpp

```diff
@@ -384,45 +385,105 @@ RewriteResponse ArithRewriter::postRewriteTranscendental(TNode t) {
       }
     }else{
+      // get the factor of PI in the argument
       Node pi_factor;
       Node pi;
-      if( t[0].getKind()==kind::PI ){
-        pi_factor = NodeManager::currentNM()->mkConst(Rational(1));
-        pi = t[0];
-      }else if( t[0].getKind()==kind::MULT && t[0][0].isConst() && t[0][1].getKind()==kind::PI ){
-        pi_factor = t[0][0];
-        pi = t[0][1];
+      Node rem;
+      std::map<Node, Node> msum;
+      if (ArithMSum::getMonomialSum(t[0], msum))
+      {
+        pi = mkPi();
+        std::map<Node, Node>::iterator itm = msum.find(pi);
+        if (itm != msum.end())
+        {
+          if (itm->second.isNull())
+          {
+            pi_factor = mkRationalNode(Rational(1));
+          }
+          else
+          {
+            pi_factor = itm->second;
+          }
+          msum.erase(pi);
+          if (!msum.empty())
+          {
+            rem = ArithMSum::mkNode(msum);
+          }
+        }
+      }
+      else
+      {
+        Assert(false);
       }
+
+      // if there is a factor of PI
       if( !pi_factor.isNull() ){
         Trace("arith-tf-rewrite-debug") << "Process pi factor = " << pi_factor << std::endl;
         Rational r = pi_factor.getConst<Rational>();
-        Rational ra = r.abs();
+        Rational r_abs = r.abs();
         Rational rone = Rational(1);
-        Node ntwo = NodeManager::currentNM()->mkConst( Rational(2) );
-        if( ra > rone ){
+        Node ntwo = mkRationalNode(Rational(2));
+        if (r_abs > rone)
+        {
           //add/substract 2*pi beyond scope
-          Node ra_div_two = NodeManager::currentNM()->mkNode( kind::INTS_DIVISION, NodeManager::currentNM()->mkConst( ra + rone ), ntwo );
+          Node ra_div_two = NodeManager::currentNM()->mkNode(
+              kind::INTS_DIVISION, mkRationalNode(r_abs + rone), ntwo);
           Node new_pi_factor;
           if( r.sgn()==1 ){
             new_pi_factor = NodeManager::currentNM()->mkNode( kind::MINUS, pi_factor, NodeManager::currentNM()->mkNode( kind::MULT, ntwo, ra_div_two ) );
           }else{
             Assert( r.sgn()==-1 );
             new_pi_factor = NodeManager::currentNM()->mkNode( kind::PLUS, pi_factor, NodeManager::currentNM()->mkNode( kind::MULT, ntwo, ra_div_two ) );
           }
-          return RewriteResponse(REWRITE_AGAIN_FULL, NodeManager::currentNM()->mkNode( kind::SINE,
-                                                       NodeManager::currentNM()->mkNode( kind::MULT, new_pi_factor, pi ) ) );
-        }else if( ra == rone ){
-          return RewriteResponse(REWRITE_DONE, NodeManager::currentNM()->mkConst(Rational(0)));
-        }else{
+          Node new_arg =
+              NodeManager::currentNM()->mkNode(kind::MULT, new_pi_factor, pi);
+          if (!rem.isNull())
+          {
+            new_arg =
+                NodeManager::currentNM()->mkNode(kind::PLUS, new_arg, rem);
+          }
```

Comment:
Shouldn't you have the creation of ``new_arg`` on an ``if ... else ...``? If ``!rem.isNull()`` holds you have a spurious node construction above.

Reply:
new_arg is being replaced by itself + rem here, so the Node construction is not spurious.